lib/autorestPlugin/pluginHost.ts

```diff
@@ -2,150 +2,204 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-Object.defineProperty(exports, "__esModule", { value: true });
-const vscodeJsonRpc = require("vscode-jsonrpc");
-const linq = require('linq');
-const jsonPath = require('jsonpath');
-const yaml = require("js-yaml");
-const utils = require("../util/utils");
-import log = require('../util/logging');
-const SpecValidator = require('../validators/specValidator');
-const extensionBase = require('@microsoft.azure/autorest-extension-base');
-const openAPIDocUrl = "https://github.com/Azure/oav";
 
-exports = module.exports;
+// import vscodeJsonRpc = require("vscode-jsonrpc")
+import * as linq from 'linq'
+import * as jsonPath from 'jsonpath'
+import * as yaml from "js-yaml"
+import * as utils from "../util/utils"
+import { log } from '../util/logging'
+import { SpecValidator } from '../validators/specValidator'
+import * as extensionBase from '@microsoft.azure/autorest-extension-base'
+import { IAutoRestPluginInitiator } from '@microsoft.azure/autorest-extension-base/dist/lib/extension-base';
 
-const extension = new extensionBase.AutoRestExtension();
-const modelValidatorPluginName = "model-validator";
-const modelValidationCategory = "ExampleModelViolation";
+const openAPIDocUrl = "https://github.com/Azure/oav"
+
+const extension = new extensionBase.AutoRestExtension()
+const modelValidatorPluginName = "model-validator"
+const modelValidationCategory = "ExampleModelViolation"
 
 class FormattedOutput{
   constructor(
-    public channel: any,
-    public details: any,
-    public code: any,
-    public text: any,
-    public source: any,
-    _?: any) {
+    public readonly channel: string,
+    public readonly details: any,
+    public readonly code: any[],
+    public readonly text: string,
+    public readonly source: any[]) {
   }
 }
 
 /**
  * Returns a promise with the examples validation of the swagger.
  */
-function analyzeSwagger(swaggerFileName: any, autoRestApi: any) {
-  return autoRestApi.ReadFile(swaggerFileName).then((swaggerFile: any) => {
-    const swagger = yaml.safeLoad(swaggerFile);
-    return exports.openApiValidationExample(swagger, swaggerFileName).then(function (exampleValidationResults: any) {
-      for (const result of exampleValidationResults) {
-        autoRestApi.Message({ Channel: result.channel, Text: result.text, Details: result.details, Key: result.code, Source: result.source });
-      }
-      // console.error(JSON.stringify(exampleValidationResults, null, 2));
-    });
-  });
+async function analyzeSwagger(swaggerFileName: string, autoRestApi: extensionBase.Host)
```

Comment:
>analyzeSwagger [](start = 15, length = 14)

i was thinking of making all the functions arrow functions i.e. ``const analyzeSwagger = async (params) => {}